### PR TITLE
Search support for the implicitly declared class

### PR DESCRIPTION
**Pull request by subyssurendran666**
## Author checklist

- [x] I have thoroughly tested my changes
- [x] The change is following the [coding conventions](https://wiki.eclipse.org/Platform/How_to_Contribute#Coding_Conventions)
- [x] I have signed the [Eclipse Contributor Agreement (ECA)](https://www.eclipse.org/legal/ECA.php)

### CODE REVIEW

File: org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/model/JavaSearchImplicitTypeDeclaration.java [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+import org.eclipse.jdt.core.WorkingCopyOwner;
+import org.eclipse.jdt.core.search.IJavaSearchScope;
+import org.eclipse.jdt.core.search.ReferenceMatch;
+import org.eclipse.jdt.core.search.SearchEngine;
+import org.eclipse.jdt.core.search.SearchMatch;
+import org.eclipse.jdt.core.search.TypeReferenceMatch;
+
+import junit.framework.Test;
+
+public class JavaSearchImplicitTypeDeclaration extends JavaSearchTests {
```
Comment (mpalat) [inline org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/model/JavaSearchImplicitTypeDeclaration.java:20]: Please add "Test" at the end ie JavaSearchImplicitTypeDeclarationTests

File: org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/model/JavaSearchImplicitTypeDeclaration.java [hunk 2]
```diff
@@ -0,0 +1,144 @@
+package org.eclipse.jdt.core.tests.model;
```
Comment (mpalat) [inline org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/model/JavaSearchImplicitTypeDeclaration.java:1]: Please add the header

**Review by mpalat — APPROVED** [review]
Please incorporate the review comments and submit. Thanks!

File: org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/model/JavaSearchImplicitTypeDeclaration.java (hunk 1 above)
Comment (subyssurendran666) [reply org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/model/JavaSearchImplicitTypeDeclaration.java]: Done

File: org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/model/JavaSearchImplicitTypeDeclaration.java (hunk 2 above)
Comment (subyssurendran666) [reply org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/model/JavaSearchImplicitTypeDeclaration.java]: Done

**Comment by subyssurendran666** [comment]
> @subyssurendran666 I don't see the review comments resolved?

Done. Could you please re-review the changes ?

**Pull request merged**